### PR TITLE
Document Enable trait types

### LINKED ISSUE #631: Enable: document trait types
Enable defines a number of trait types in various places.  Many of these are intended for external use.  We should document them, if only to mention their existence and briefly state their purpose.

### PR DESCRIPTION
**Pull request by jwiggins**
Closes #631

### CODE REVIEW

File: docs/source/enable/traits.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
+Enable Custom Traits
+====================
+Enable defines several `trait types <https://docs.enthought.com/traits>`_ which
```
Comment (aaronayres35) [inline docs/source/enable/traits.rst:3]: Maybe link to the custom traits section of the traits docs?
https://docs.enthought.com/traits/traits_user_manual/custom.html

I actually am unsure which would make more sense, so feel free to leave if you prefer the home page

**Review by aaronayres35 — APPROVED** [review]
LGTM

Most of the links appear to be dead.  Trait definitions are strange though so I think they may not even have api docs correctly generated.  
see https://docs.enthought.com/enable/api/enable.api.html#enable-trait-types

seems like the markers do so they could be linked, but that is probably unnecessary

File: docs/source/enable/traits.rst (hunk 1 above)
Comment (jwiggins) [reply docs/source/enable/traits.rst:3]: I'll leave it like this

**Comment by jwiggins** [comment]
Thanks for the review

**Pull request merged**